penpot/penpot
# :recycle: Refactor profile props

This PR mainly consists on flowing things:

- Add separated (namespaced) storage implementation for separate user and global values. User values are automatically cleared on logout but global are persisted.
- Use storage/user for almost all the things where storage is used previously
- Move several props stored on user profile to the storage/global namespace on the browser
- Add schema validation for known profile props.
- Add minor internal refactor on schema ns (removes unused code and makes private several functions)

From now, all props on profile should be added to the schema for proper validation and decoding.

How to test:

- Profile post registration steps should work correctly (they set many props)
- Not default team hero on dashboard project page, the collapsed state is persisted among refresh and logout
- Recent colors should be persisted for restarts but reset on logout
- Builtin templates collapsed state should persist restart and logout
